source/frontend/src/features/leases/shared/propertyPicker/LeaseUpdatePropertySelector.tsx

```diff
@@ -258,42 +265,40 @@ export const LeaseUpdatePropertySelector: React.FunctionComponent<
   };
 
   const saveFile = async (file: ApiGen_Concepts_Lease) => {
-    try {
-      const response = await updateLeaseProperties.execute(file, []);
-
-      formikRef.current?.setSubmitting(false);
-      if (isValidId(response?.id)) {
-        if (file.fileProperties?.find(fp => !fp.property?.address && !fp.property?.id)) {
-          toast.warn(
-            'Address could not be retrieved for this property, it will have to be provided manually in property details tab',
-            { autoClose: 15000 },
-          );
-        }
-        formikRef.current?.resetForm();
-        await getCompleteLease();
-        mapMachine.refreshMapProperties();
-        pathSolver.showFile('lease', lease.id);
-      }
-    } catch (e) {
-      if (axios.isAxiosError(e) && (e as AxiosError).code === '409') {
-        setShowAssociatedEntityWarning(true);
-      } else if (e?.response?.data?.type === 'BusinessRuleViolationException') {
-        setShowSaveConfirmModal(false);
+    return withUserOverride(
+      (userOverrideCodes: UserOverrideCode[]) => {
+        return updateLeaseProperties.execute(file, userOverrideCodes).then(async response => {
+          formikRef.current?.setSubmitting(false);
+          if (isValidId(response?.id)) {
+            if (file.fileProperties?.find(fp => !fp.property?.address && !fp.property?.id)) {
+              toast.warn(
+                'Address could not be retrieved for this property, it will have to be provided manually in property details tab',
+                { autoClose: 15000 },
+              );
+            }
+            formikRef.current?.resetForm();
+            await getCompleteLease();
+            mapMachine.refreshMapProperties();
+            pathSolver.showFile('lease', lease.id);
+          }
+        });
+      },
+      [],
+      (axiosError: AxiosError<IApiError>) => {
```

Comment:
question: the old logic would show a custom modal with text: 
> This property can not be removed from the file. This property is related to one or more entities in the file, only properties that are not linked to any entities in the file can be removed.

But the hotfix removes the only call to `setShowAssociatedEntityWarning(true);` - shouldn't we remove the custom modal as well? Also how is this change still supporting that flow? Are doing the check in the backend?

Reply:
Removed the modal. This error appears to be legacy and is only thrown in one area of the backend - when a payee is removed from a lease. Also, this error was not working before as the axiosError.code would never be 409.